### DISCUSSION TITLE
Code annotations without popups

### DISCUSSION THREAD

**Discussion by dogsbay** [squidfunk/mkdocs-material#5289] [Q&A] [answered]
Would it be possible to have an option to render annotations as normal lists after codeblocks, as opposed to popups? 

There are a few reasons for asking for this:

- For wide codeblocks, sometimes the annotations are off to the rigth and not immediately visisible without scrolling, so users can miss them entirely
- Improved accessibility - no need for extra clicks to view popups
- More familiar for users coming from DocBook/Asciidoc where there is a numbered icon in the code block, and then the list item with  numbered icon afterward the code block rather than as popup
- Better facilitates nested content in the list item - I have seen https://github.com/squidfunk/mkdocs-material/discussions/3903 but I am not sure a popup is the best interface when a list item has significant nested structures, such as images/lists/tables/etc
- You can see more than one annotation at a time

So basically, I would just like numbered icons in the codeblock, and the following list, but the list treated as a regular list otherwise - maybe there is a way to get this already that I am not aware of?

**Comment by squidfunk** [comment, marked as answer]
Thanks for suggesting. I understand the use case and have seen it on AsciiDoc, but it's really beyond the scope of this project. This can actually be very easily implemented with customization, similar to how code annotations do it: detect a specific marker in `code` elements that are succeeded by a list element, and then apply styling accordingly ☺️

**Reply by dogsbay** [reply]
Thanks, Martin. I will have to investigate a bit more, as I have never done a customization.
